### PR TITLE
Move JDK11 ITs to cron stage

### PR DESCRIPTION
**Pull request by abhishekagarwal87**
This PR moves some of the ITs for JRE 11 runtime to the cron stage. The reason is simple - we have a lot of tests and our PR checks take too long to finish. This would improve the situation slightly. I have kept the following ITs around
 - query integration test (so we at least check the whole ingestion and querying end-to-end on runtime 11)
 - security integration test, leadership integration test, query integration test (mariaDB) - I kept these around since they involve another service like a zookeeper. There are no unit tests with the MariaDB driver. The same goes for TLSCertificateChecker.

This PR has:
- [x] been self-reviewed.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by gianm** [comment]
When do the cron tests run and how can we see the results? Is there a way to run them on-demand on release branches? (Running "extra bonus" tests on release candidates seems like the best use of them.)

Please forgive me for these basic questions; I'm just not familiar with how the cron tests work.

File: .travis.yml [hunk 1]
```diff
@@ -744,21 +750,25 @@ jobs:
...
 
     - <<: *integration_compaction_tests
       name: "(Compile=openjdk8, Run=openjdk11) compaction integration test"
       jdk: openjdk8
+      stage: cron
       env: TESTNG_GROUPS='-Dgroups=compaction' JVM_RUNTIME='-Djvm.runtime=11' USE_INDEXER='middleManager'
 
     - <<: *integration_tests
       name: "(Compile=openjdk8, Run=openjdk11) other integration test"
       jdk: openjdk8
```
Comment (kfaraz) [inline .travis.yml:770]: Is this correct? This entry seems to be for jdk8.

Comment (abhishekagarwal87) [reply .travis.yml:770]: yes. druid gets compiled and built with JDK 8 but then tests are run on JRE 11.

**Comment by abhishekagarwal87** [comment]
> When do the cron tests run and how can we see the results? Is there a way to run them on-demand on release branches? (Running "extra bonus" tests on release candidates seems like the best use of them.)
> 
> Please forgive me for these basic questions; I'm just not familiar with how the cron tests work.

I forgive you :) 
The cron tests run daily on the master branch. It can be configured in the https://app.travis-ci.com/github/apache/druid/settings. 
The notifications for success/failures are sent to dev@druid.apache.org. For some reason, I don't see a daily email. If you search "[CRON]" in your email, you can see these notifications. 

I don't see a way to run them on-demand on release branches. We can however make a slight change in `.travis.yml` so the cron stage becomes a regular third stage.

**Comment by abhishekagarwal87** [comment]
I could also set a condition such that cron jobs would always run on master and release branches but not on pull requests.

**Comment by gianm** [comment]
> I could also set a condition such that cron jobs would always run on master and release branches but not on pull requests.

Seems like running them automatically on release branches would be good; that way, it makes life easier for release managers. They would be able to backport things without waiting for CI to pass on each PR, since they could look at the release-branch results after backports are done.

**Comment by gianm** [comment]
> I could also set a condition such that cron jobs would always run on master and release branches but not on pull requests.

Is this something that would be done in `.travis.yml` in this PR? If so, it sounds good to me, for the reasons in my prior comment.

**Comment by gianm** [comment]
Final note: please also update `distribution/asf-release-process-guide.md` so release managers know to check these extra tests.

**Comment by abhishekagarwal87** [comment]
@gianm - I have made the changes you suggested. PTAL.

File: distribution/asf-release-process-guide.md [hunk 2]
```diff
@@ -113,9 +113,11 @@ For docs, please make sure to add any relevant redirects in `website/redirects.j
 
 ### Release branch hygiene
 
-The only additions to the release branch after branching should be bug fixes, which should be back-ported from the master branch, via a second PR, not with a direct PR to the release branch. Bug fix release branches may be initially populated via cherry-picking, but it is recommended to leave at least 1 commit to do as a backport PR in order to run through CI. (Note that CI is sometimes flaky for older branches).
+The only additions to the release branch after branching should be bug fixes, which should be back-ported from the master branch, via a second PR or a cherry-pick, not with a direct PR to the release branch.  
 
-Once all issues and PRs that are still tagged with the release milestone have been merged, closed, or removed from the milestone, the next step is to put together a release candidate.
+Release manager must also ensure that CI is passing successfully on the release branch. Since CI on branch can contain additional tests such as ITs for different JVM flavours. (Note that CI is sometimes flaky for older branches).
```
Comment (gianm) [inline distribution/asf-release-process-guide.md:118]: Could you please include some instruction about how to ensure that CI passes successfully on the branch? Like, a link or a description of what to click where?

File: .travis.yml [hunk 3]
```diff
@@ -53,7 +53,8 @@ install: ./check_test_suite.py && travis_terminate 0  || echo 'Running Maven ins
 # 2. Tests - phase 2
 # 3. cron
 #
-# The cron type only runs jobs that are marked with stage cron.
+# The cron type only runs jobs that are marked with stage cron. The ron stage also runs alongside
```
Comment (gianm) [inline .travis.yml:56]: "cron stage" (spelling)

File: .travis.yml [hunk 4]
```diff
@@ -65,7 +66,7 @@ stages:
   - name: Tests - phase 2
     if: type != cron
   - name: cron
-    if: type = cron
+    if: type = cron OR (type != pull_request AND branch != master)
```
Comment (gianm) [inline .travis.yml:69]: https://docs.travis-ci.com/user/conditions-v1 says the possible `type` are `push`, `pull_request`, `api`, `cron`. So these tests will now additionally run for nonmaster branches if the event type is `push` or `api`. I guess `api` is for manually-triggered tests on branches. What's `push` for?

Comment (abhishekagarwal87) [reply .travis.yml:69]: I am guessing but it must be triggered for any commit that is `pushed` to the branch. I could not explicit documentation but that definition seems most plausible.

Comment (gianm) [reply .travis.yml:69]: Ok, let's try this and see if it does what we want.

**Review by gianm — APPROVED** [review]

**Pull request merged**